src/Workspaces/Core/Desktop/Workspace/FileTextLoader.cs

```diff
@@ -7,11 +7,21 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis
 {
+    internal static class FileTextLoaderOptions
+    {
+        // File Threshold 100MB
+        [ExportOption]
+        internal static readonly Option<long> FileLengthThreshold = new Option<long>(nameof(FileTextLoaderOptions), nameof(FileLengthThreshold), defaultValue: 100 * 1024 * 1024,
```

Comment:
I don't know what FileLengthThreshold means.  Can we comment this.  What happens at 100MB?  Do we simply ignore hte file entirely?  Do we have the document, but we truncate it at 100MB?  Do we have the document, but treat its contents as empty?